File: 7.cpp
#include<iostream>
using namespace std;
/* 素数判定*/
int isPrime(int x) {
	int i;
	if ( x < 2 ) return 0;
	else if ( x == 2 ) return 1; /* 2 は素数*/
	if ( x % 2 == 0 ) return 0; /* 偶数は素数ではない*/
	for ( i = 3; i*i <= x; i+=2 ) { /* i が x の平方根以下の間*/
	if ( x % i == 0 ) return 0;
	}
	return 1;
}

int main() {
	int cnt=0;
	for(long long i=1;cnt <10001;i++){
		if(isPrime(i)) cnt++;
		if(cnt==10001) cout << i << endl;
	}
}

File: 7.py
prime=[0]*100001
isPrime=[True]*1000001
def sieve(n):
	p=0
	isPrime[0]=isPrime[1]=False
	for i in range(2,n+1):
		if isPrime[i]==True:
			prime[p]=i
			p+=1
			for j in range(2*i,n+1,i) :
				isPrime[j]=False

sieve(200000)
print prime[10001]